controllers/machinehealthcheck_controller.go

```diff
@@ -445,3 +462,21 @@ func machineNames(machines []*clusterv1.Machine) []string {
 	}
 	return result
 }
+
+func getMaxUnhealthy(mhc *clusterv1.MachineHealthCheck) (int, error) {
+	if mhc.Spec.MaxUnhealthy == nil {
+		// This value should be defaulted, but if not, 100% is the default
+		return int(mhc.Status.ExpectedMachines), nil
```

Comment:
slight preference for returning an error here if somehow that value isn't set rather than silently defaulting in the controller

Reply:
That would change the existing logic, currently we just return allowed if maxUnhealthy is not set, so this 100% is acting as a proxy for that allowance.

I could return an error and then check if it's a particular "not set" error and behave appropriately then, WDYT?
```
maxUnhealthy, err := getMaxUnhealthy(mhc)
if err == MaxUnhealthyNotSetErr {
  return true, nil 
} else if err != nil {
  return false, err
}
```

I don't know if this really adds that much though, perhaps a better explanation of why we default here would be better?

Reply:
Updated as per your suggestion. I've added a TODO comment above the code and will follow up on this with a PR to do that change. I don't think it's too major but I'd prefer to keep any discussion about a change of logic to a separate PR so that we don't hold up this UX change